### ISSUE TITLE
vasm v's devpac

### ISSUE THREAD

**Issue by marc365** [AmosFactory/AmosProfessional#26]
[vasm](http://sun.hasenbraten.de/vasm/) is maintained, the date on the last release is the 14th of Feb 2017

I read that there is a devpac compatibility mode, and with a few minor tweaks to +w.s I got an exact copy of AMOS.library assembled and working with vasm 1.7h - I believe it's an exact copy, it's the same size (and I did delete and recreate it to check)

this is very exciting news and up for discussion:

vasmm68k_mot -Fhunkexe -o libs:AMOS.library -nosym -nocase -devpac +w.s

**Comment by mness1978** [comment]
This is great news!

I was thinking about this myself a couple of days ago and was thrilled to hear that you already did the testing.  Did you do a diff on the two AMOS.library files to determine if it is indeed an exact copy?  If they are it would be awesome news.

Even if they aren't exactly the same, I think we should seriously consider migrating to the vasm/vlink tool chain. I believe all optimizations are turned off in Devpac compatibility mode, so eventually moving away from the compatibility mode altogether and going straight with vasm may be prudent.

Also, using vasm would make possible cross assembling.  This would be very handy for me.

**Comment by SamuraiCrow** [comment]
I agree with Mike.  Not only is VAsm better than DevPac for optimization, it is freeware with C source available for noncommercial use.  I don't own DevPac, myself, so the freeware software is legally safer for my use.

**Comment by marc365** [comment]
cool, can we look into a replacement for the Monam as well, there are a few monitors on Aminet - there is no rush to get this though...

**Comment by SamuraiCrow** [comment]
http://de.aminet.net/aminet/dev/asm/BarflyDisk2_00.lha has a debugger that is supposed to be very good called BDebug.  Readme is at http://de.aminet.net/aminet/dev/asm/BarflyDisk2_00.readme if other information is needed.

**Comment by marc365** [comment]
it hasn't failed on me yet - even though a deeper comparison of the binaries show they are different, it hasn't crashed and everything seems normal.

and have built AMOSPro successfully, it's a few bytes smaller and runs perfectly...

I will branch and complete the process, upload it for testing and to confirm.

**Comment by marc365** [comment]
APCmp has many branch destination out of range errors generated by vasm.

**Comment by marc365** [comment]
+lib.s (AMOSPro.Lib) isn't working with vasm yet:

warning 2061 in line 26642 of "+lib.s": division by zero
>L284F4	DIVU	#0,D7 

warning 2018 in line 25144 of "+lib.s": double registers in list
movem.l	a2-a2/d2-d7,-(sp)

warning 2018 in line 25283 of "+lib.s": double registers in list
movem.l	(sp)+,a2-a2/d2-d7

**Comment by marc365** [comment]
nope - they are just warnings, turning off -nocase makes this build and execute correctly - so they all work - I'll have a play to confirm it.

**Comment by marc365** [comment]
this first stage is complete and merged (thank you Michael for reviewing so quickly), eventually the genam binary and 'a' scripts can be removed, currently the source still builds with both vasm and devpac. the W.s is case sensitive, I did some work to convert then, but it was very boring so I used -nocase instead.

using -nocase on files that don't need it makes the executable fail, at a guess the jump tables were wrong, but the ones that have it need it, for example there is a references in equ.s to Dosinfo (lowercase i) and in the includes there is a refrence DosInfo (uppercase i) - the actual refernce being completely different - the 'v' scripts only use -nocase if it is needed. 

next stage would be an experiment with turning off devpac compatibility so see what the work load of changes would be, maybe we can do this in stages with the source and it won't be so risky, but this is a job for later.
